server/src/main/java/org/elasticsearch/action/support/replication/ReplicationOperation.java

```diff
@@ -132,6 +132,13 @@ private void handlePrimaryResult(final PrimaryResultT primaryResult) {
             if (logger.isTraceEnabled()) {
                 logger.trace("[{}] op [{}] completed on primary for request [{}]", primary.routingEntry().shardId(), opType, request);
             }
+
+            pendingActions.incrementAndGet();
+            replicasProxy.onPrimaryOperationComplete(
+                replicaRequest,
+                ActionListener.wrap(ignored -> decPendingAndFinishIfNeeded(), e -> { finishAsFailed(e); })
```

Comment:
nit: I think we can write this? (haven't tried tho)

```suggestion
                ActionListener.wrap(ignored -> decPendingAndFinishIfNeeded(), this::finishAsFailed)
```